xapian-letor/ranker/ranker.cc

```diff
@@ -354,6 +354,9 @@ Ranker::labelcomparer(const FeatureVector & firstfv, const FeatureVector& second
 std::vector<Xapian::docid>
 Ranker::rank(const Xapian::MSet & mset, const string & model_key, const Xapian::FeatureList & flist)
 {
+    if (mset.empty()) {
+	throw Xapian::InvalidArgumentError("Empty MSet. No documents could be retrieved with the given Query.\n");
```

Comment:
This would be a lot easier to get right in #156 where you could just `return` from the method is `mset.empty()`. Throwing an exception is the wrong thing, because it means that every client of `Ranker::rank()` has to catch it and do something useful with it — and the something useful in this case is to _ignore the exception_ and do any normal processing on an MSet (which may include checking if it's empty).